compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -312,8 +311,21 @@ impl<'context> Elaborator<'context> {
             self.in_unconstrained_fn = true;
         }
 
-        let func_meta = self.extract_meta(&function, id);
+        let func_meta = self
+            .interner
+            .func_meta
+            .get(&id)
+            .expect("FuncMetas should be declared before a function is elaborated")
+            .clone();
+
+        for (parameter, param2) in function.def.parameters.iter().zip(&func_meta.parameters.0) {
+            let definition_kind = DefinitionKind::Local(None);
+            self.elaborate_pattern(parameter.pattern.clone(), param2.1.clone(), definition_kind);
```

Comment:
This is repeated work since parameters are already elaborated when creating the FuncMeta but we don't have an easy way of reintroducing existing items into scope yet.

Reply:
nit it would be nice to note this in comments

Reply:
I ended up removing this in jf/elaborator-fixes8 so I'll keep this for now until that PR is merged which removes it anyway